### PR TITLE
Performance: Optimizing the Rntbdtoken codec look-up O(n) -> O(1)

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Current implementation doing a for-loop, i,e, for any fromId() invocation results in 16times CPU loop.
New implementation pre-caches them and used index lookup.

JHM results below

|Benchmark          |     Mode  | Cnt      |    Score    |     Error | Units |
|-------------------|-----------|----------|-------------|-----------|-------|
|MyBenchmark.current(last-entry)   | thrpt |  25 |  22000729.476 | ±  444587.766 | ops/s |
|MyBenchmark.optimized(last-entry) | thrpt |  25 | 500388199.392 | ± 1068858.996 | ops/s |
|MyBenchmark.current(first-entry)   | thrpt |  25 |  35150618.211 | ±  184202.541 | ops/s |
|MyBenchmark.optimized(first-entry) | thrpt |  25 | 505749163.599 | ± 1320707.249 | ops/s |

### CODE REVIEW

**Comment by kirankumarkolli** [comment]
Haven't ran the test E2E, micro-benchmark might be best to such scenario no?

**Comment by kirankumarkolli** [comment]
> do you see any perf difference (when testing e2e)?

Haven't ran the test E2E, micro-benchmark might be best to such scenario no?

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdTokenTypeTests.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.azure.cosmos.implementation.directconnectivity.rntbd;
+
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class RntbdTokenTypeTests {
+
+    @Test(groups = { "unit" })
+    public void testAllTokenTypes() {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdTokenTypeTests.java:10]: we don't use "test" as prefix nor suffix as a test code style.
```suggestion
    public void allTokenTypes() {
```

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdTokenTypeTests.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.azure.cosmos.implementation.directconnectivity.rntbd;
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdTokenTypeTests.java:1]: header filing is missing. please add.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdTokenType.java [hunk 3]
```diff
@@ -42,6 +42,24 @@
     Invalid((byte)0xFF, RntbdNone.codec);             // no data
 
     // region Implementation
+    private static final RntbdTokenType[] allTokens = getAllTokens();
+
+    private static  RntbdTokenType[] getAllTokens() {
+        final int maxByteValue = 0xFF + 1;
+        final RntbdTokenType[] allPossibleTokens = new RntbdTokenType[maxByteValue]; // one byte RNTBD limit
+        for(int i=0; i< maxByteValue; i++) {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdTokenType.java:50]:
```suggestion
        for(int i = 0; i < maxByteValue; i++) {
 ```

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdTokenTypeTests.java (hunk 2 above)
Comment (kirankumarkolli) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdTokenTypeTests.java:1]:
```suggestion
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.azure.cosmos.implementation.directconnectivity.rntbd;
```

**Pull request merged**